closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -44,17 +141,192 @@ typedef struct OpenSSLMDContext {
 static void printErrors(void) {
     unsigned long errCode = 0;
 
-    fprintf(stderr, "An error occurred\n");
-    while(0 != (errCode = ERR_get_error()))
+    fprintf(stderr, "An OpenSSL error occurred\n");
+    while(0 != (errCode = (*OSSL_get_error)()))
     {
         char err_str[120];
-        ERR_error_string_n(errCode, err_str, (sizeof(err_str) / sizeof(char)));
-        fprintf(stderr, "Generating error message\n" );
+        (*OSSL_error_string_n)(errCode, err_str, (sizeof(err_str) / sizeof(char)));
         fprintf(stderr, "%s\n", err_str);
     }
     fflush(stderr);
 }
 
+/*
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    loadCrypto
+ * Signature: ()I
+ */
+JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+  (JNIEnv *env, jclass thisObj){
+
+    void *handle = NULL;
+    typedef const char* OSSL_version_t(int);
+
+     // Determine the version of OpenSSL.
+    OSSL_version_t* OSSL_version;
+    const char * openssl_version;
+    int ossl_ver;
+
+    // Load OpenSSL Crypto library
+    handle = load_crypto_library();
+    if (NULL == handle) {
+        //fprintf(stderr, " :FAILED TO LOAD OPENSSL CRYPTO LIBRARY\n");
+        //fflush(stderr);
+        return -1;
+    }
+
+    // Different symbols are used by OpenSSL with 1.0 and 1.1.
+    // The symbol 'OpenSSL_version' is used by OpenSSL 1.1 where as
+    // the symbol "SSLeay_version" is used by OpenSSL 1.0.
+    // Currently only openssl 1.0.2 and 1.1.0 and 1.1.x are supported.
```

Comment:
'and 1.1.0' is redundant (here and in other comments below)

Reply:
Done